### DISCUSSION TITLE
Tracking usage

### DISCUSSION THREAD

**Discussion by Marquo1** [TillFleisch/ESPHome-Philips-Smart-Coffee#70] [Ideas]
Hi, this is a very cool project 😁 I'm considering pulling the trigger, but I thought I'll ask first:

I'd like to use this to know when the coffee machine needs maintenance.

According to info I found, the maintenance schedule is:
- use Oil removal tablet - every 500 cups
- Grease brew group - every 500 cups
- change AquaPure filter - every 625 cups

To do this I need to register both cups made by using the screen and by HA. I imagine using the status sensor and utility meter to count the cups.
Will the status sensor provide enough information for this?

Edit:
Changed the frequency based on markings inside the machine.

**Comment by TillFleisch** [comment]
> change AquaPure filter

In this case the machine complains itself. There is a separate led on the panel just for this purpose. This repo currently does not support the aqua/calc clean led

Counting cups is possible with the Status sensor. It reports the current state including a 'brewing' (beverage specific) state. It's currently a bit flakey, there is an open issue iirc. (I will eventually get around to fixing this.)

Wether the cups are ordered from the panel or 'through HA' does not make a difference for the status sensor.

Power utilisation alone could be a sufficient indicator

**Reply by Marquo1** [reply]
Thanks for answering! I realise that. And I don't think support for it is really needed. I'm thinking of it as an early warning to buy one in time.
Speaking of, I wonder if the machine counts the cups or just starts complaining after certain time. 🤔

Monitoring power consumption can work! But I also want to turn the machine on before I come downstairs in the morning and reproduce features of higher end machines (coffee presets).
So this repo does it all for me ❤️

**Comment by quack3d** [comment]
> According to info I found, the maintenance schedule is:
> 
> * use Oil removal tablet - every 500 cups
> * Grease brew group - approx. 600 cups
> * change AquaPure filter - every 625 cups

Where did you get that info btw? I've just found an estimate of days. 7 days for cleaning the insides and the brewer, 30 days for tablet cleaning and 90 days for applying grease.

**Reply by Marquo1** [reply]
The manual for my machine had the same, but it felt very conservative so I started digging.

[Here](https://www.usa.philips.com/c-f/XC000004177/how-to-lubricate-the-brew-group-of-my-philips-espresso-machine) is my source for lubrication frequency depending on how many cups you make per day. I extrapolated into number of cups instead. 
```
Depending on the usage of your espresso machine, see below advised frequencies:
1. If you brew 1 to 5 coffees a day, lubricate every 4 months
2. If you brew 6 to 10 coffees a day, lubricate every 2 months
3. If you brew more than 10 coffees a day, lubricate once a month
```

And [here](https://www.coffeefriend.co.uk/p/coffee-oil-removing-tablets-philips-ca6704-10/) for tablets.
```
It’s recommended to clean your coffee machine every month 
or after every 500 cups.
 ```

The recommendation for weekly cleaning of the insides makes sense, no argument there.

EDIT: 
I just found these markings on my brew section. 
![20240630_112108](https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee/assets/97259078/1f142943-62e0-4b17-863d-cf30243f8601)
